src/track/tracknumbers.cpp

```diff
@@ -0,0 +1,106 @@
+#include <QStringList>
+
+#include "track/tracknumbers.h"
+#include "util/assert.h"
+
+/*static*/ const QString TrackNumbers::kDefaultSeparator("/");
+
+//static
+std::pair<TrackNumbers, TrackNumbers::ParseResult> TrackNumbers::fromString(
+        const QString& str,
+        const QString& separator) {
+    TrackNumbers trackNumbers;
+    if (str.trimmed().isEmpty()) {
+        return std::make_pair(trackNumbers, ParseResult::EMPTY);
+    } else {
+        const QStringList splitted(str.split(separator));
+        DEBUG_ASSERT(splitted.size() > 0);
+        ParseResult parseResult;
+        if ((splitted.size() == 1) || (splitted.size() == 2)) {
+            parseResult = ParseResult::VALID;
+            bool currentValid = false;
+            int current = splitted[0].toInt(&currentValid);
+            if (currentValid) {
+                trackNumbers.setCurrent(current);
+            } else {
+                parseResult = ParseResult::INVALID;
+            }
+            if (splitted.size() > 1) {
+                bool totalValid = false;
+                int total = splitted[1].toInt(&totalValid);
+                if (totalValid) {
+                    trackNumbers.setTotal(total);
+                } else {
+                    parseResult = ParseResult::INVALID;
+                }
+            }
+            if (!trackNumbers.isValid()) {
+                parseResult = ParseResult::INVALID;
+            }
+        } else {
+            parseResult = ParseResult::INVALID;
+        }
+        return std::make_pair(trackNumbers, parseResult);
+    }
+}
+
+//static
+std::pair<QString, QString> TrackNumbers::splitString(
+        const QString& str,
+        const QString& separator) {
```

Comment:
It looks like separator is always default, so I you can just remove the parameter.